src/streamlink/plugins/pandalive.py

```diff
@@ -0,0 +1,86 @@
+import logging
+import re
+
+from streamlink.plugin import Plugin, PluginError, pluginmatcher
+from streamlink.plugin.api import validate
+from streamlink.stream.hls import HLSStream
+
+log = logging.getLogger(__name__)
+
+
+@pluginmatcher(re.compile(
+    r"https?://(?:www\.)?pandalive\.co\.kr/"
+))
+class Pandalive(Plugin):
+    API_URL = "https://api.pandalive.co.kr/v1/live/play"
+
+    _room_id_re = re.compile(r"roomid\s*=\s*String\.fromCharCode\((.*)\)")
+
+    def _get_media_code(self):
+        root = self.session.http.get(self.url, schema=validate.Schema(
+            validate.parse_html(),
+        ))
+
+        try:
+            schema = validate.Schema(
+                validate.xml_xpath_string(".//script[contains(text(), 'roomid')]/text()"),
+                str,
+                validate.transform(self._room_id_re.search),
+                validate.get(1),
+                str,
+                validate.transform(lambda s: "".join(map(lambda c: chr(int(c)), s.split(",")))),
+            )
+            return schema.validate(root)
+        except PluginError:
+            pass
+
+        if not root.text:
+            log.error("The broadcast has ended")
+
+        return None
+
+    def _get_streams(self):
+        media_code = self._get_media_code()
+
+        if not media_code:
+            return
+
+        log.debug(f"Media code: {media_code}")
+
+        data = {
+            "action": "watch",
+            "mediaCode": media_code,
+        }
+        json = self.session.http.post(self.API_URL, data=data, schema=validate.Schema(
+            validate.parse_json(), {
+                validate.optional("media"): {
+                    "title": str,
+                    "userId": str,
+                    "userNick": str,
+                    "isLive": bool,
+                },
+                validate.optional("PlayList"): {
+                    "hls2": [{
+                        "url": validate.url(),
+                    }],
+                },
+                "result": bool,
+            },
+        ))
+
+        if not json:
+            log.debug("No JSON returned from API request")
+            return
```

Comment:
`json` can't be falsey with the validation schema from above. If the schema doesn't succeed, it'll raise a `PluginError`.

Reply:
Ah, yes, thanks.